src/frequenz/sdk/actor/_power_managing/_matryoshka.py

```diff
@@ -44,12 +44,12 @@ class Matryoshka(BaseAlgorithm):
     def __init__(self, max_proposal_age: timedelta) -> None:
         """Create a new instance of the matryoshka algorithm."""
         self._max_proposal_age_sec = max_proposal_age.total_seconds()
-        self._component_buckets: dict[frozenset[int], SortedSet[Proposal]] = {}
+        self._component_buckets: dict[frozenset[int], deque[Proposal]] = {}
```

Comment:
It's not clear to me how you can replace a sorted set with a deque? I see you sort it below, but why not use a set here?

Reply:
The original `SortedSet` didn't need a `__hash__` method.  But if we add a custom one that hashes just the source_id and priority, then we are able to use a `set`.  I've made that change.